### ISSUE TITLE
Spring Cloud Context breaks Camunda property management.

### ISSUE THREAD

**Issue by Zapata** [camunda/camunda-bpm-spring-boot-starter#189]
Application properties for Camunda section are not working properly when `spring-cloud-starter` is in the classpath.
There should be some issues with the [context hierarchy](http://projects.spring.io/spring-cloud/docs/1.0.3/spring-cloud.html#_application_context_hierarchies).

As an example if I take `example-cloud`, and add configuration for admin user:
```
camunda:
  bpm:
    admin-user:
      id: admin
      password: admin
```

I got this error:
```
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'simpleApplication': Unsatisfied dependency expressed through field 'historyService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'processEngineFactoryBean' defined in class path resource [org/camunda/bpm/spring/boot/starter/CamundaBpmAutoConfiguration.class]: Unsatisfied dependency expressed through method 'processEngineFactoryBean' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'processEngineConfigurationImpl' defined in class path resource [org/camunda/bpm/spring/boot/starter/CamundaBpmConfiguration.class]: Unsatisfied dependency expressed through method 'processEngineConfigurationImpl' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'createAdminUserConfiguration': Invocation of init method failed; nested exception is java.lang.NullPointerException: missing field: camunda.bpm.admin-user.id
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:592) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:370) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1219) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:551) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:754) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:866) ~[spring-context-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:542) ~[spring-context-4.3.4.RELEASE.jar:4.3.4.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.4.2.RELEASE.jar:1.4.2.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:761) [spring-boot-1.4.2.RELEASE.jar:1.4.2.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:371) [spring-boot-1.4.2.RELEASE.jar:1.4.2.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) [spring-boot-1.4.2.RELEASE.jar:1.4.2.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1186) [spring-boot-1.4.2.RELEASE.jar:1.4.2.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1175) [spring-boot-1.4.2.RELEASE.jar:1.4.2.RELEASE]
	at org.camunda.bpm.spring.boot.example.cloud.SimpleApplication.main(SimpleApplication.java:29) [classes/:na]
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'processEngineFactoryBean' defined in class path resource [org/camunda/bpm/spring/boot/starter/CamundaBpmAutoConfiguration.class]: Unsatisfied dependency expressed through method 'processEngineFactoryBean' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'processEngineConfigurationImpl' defined in class path resource [org/camunda/bpm/spring/boot/starter/CamundaBpmConfiguration.class]: Unsatisfied dependency expressed through method 'processEngineConfigurationImpl' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'createAdminUserConfiguration': Invocation of init method failed; nested exception is java.lang.NullPointerException: missing field: camunda.bpm.admin-user.id
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:749) ~[spring-beans-4.3.4.RELEASE.jar:4.3.4.RELEASE]
```

The configuration values seems to be ignored.
If you run the main `SimpleApplication`, the workflow is deployed, even if there is `camunda.bpm.auto-deployment-enabled: false`.

Without `spring-cloud-starter`everything works well.

I'll try to investigate deeper. I'll keep you posted.

**Comment by Zapata** [comment]
If I remove [this line](https://github.com/camunda/camunda-bpm-spring-boot-starter/blob/master/extension/starter/src/main/resources/META-INF/spring.factories#L7), it seems better.

It was introduced [here](https://github.com/camunda/camunda-bpm-spring-boot-starter/commit/2efc6f1eaca6cd247133f43eb74f44f7e9515664).

I don't see any reason to add it, do I miss something?

**Comment by pingunaut** [comment]
Similar problem here. Camunda seems to ignore configuration properties. They are all null

**Comment by gordonwho** [comment]
What is the work around?

**Comment by lldata** [comment]
I just lost a day on this issue, before discovering this thread...

I tried upgrading to 2.1.0-SNAPSHOT, but my "camunda.bpm.database.schema-update" property are still ignored in SpringProcessEngineConfiguration. 

As a workaround: Setting the property as a -D JVM flag or using bootstrap.properties, seems to work.

Is there another workaround/fix for this? I'd like to be able to trust my property files again ;-)

**Comment by gordonwho** [comment]
@lldata Can you provide some samples to the approaches you provided?

**Comment by lldata** [comment]
@gordonwho simply create a file called bootstrap.yaml and copy the camunda.bpm settings from application.yaml to that file.

Or when you start your program you can provide the flag that give you trouble as JVM system properties e.g. -Dcamunda.bpm.database.schema-update=create-drop

https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-external-config

**Comment by jangalinski** [comment]
tried to reproduce, suppose it's fixed with #206
